installer_scripts/install_centos7.sh

```diff
@@ -2,142 +2,171 @@
         case $yn in
             [Yy]* ) 
             echo "Installing GoLang."
-            wget https://golang.org/dl/go1.17.2.linux-amd64.tar.gz &>> mylog.log
+            wget https://golang.org/dl/${GO_VERSION}.linux-amd64.tar.gz &>> migration_installer.log
             if [ $? -ne 0 ]; then
-                echo "GoLang not installed! Check $HOME/mylog.log for more details."
+                echo "GoLang not installed! Check migration_installer.log for more details."
                 exit
             fi
-            sudo tar -C /usr/local -xzf go1.17.2.linux-amd64.tar.gz &>> mylog.log
+            sudo tar -C /usr/local -xzf ${GO_VERSION}.linux-amd64.tar.gz &>> migration_installer.log
             if [ $? -ne 0 ]; then
-                echo "GoLang not installed! Check $HOME/mylog.log for more details."
+                echo "GoLang not installed! Check migration_installer.log for more details."
                 exit
             fi
-            rm go1.17.2.linux-amd64.tar.gz
-            echo 'export PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc
-            export PATH=$PATH:/usr/local/go/bin
-            echo 'export GOPATH=$HOME/go' >> ~/.bashrc
-            export GOPATH=$HOME/go
-            echo 'export PATH=$PATH:$GOPATH/bin' >> ~/.bashrc
-            export PATH=$PATH:$GOPATH/bin
-            echo "GoLang Installed."
+            rm ${GO_VERSION}.linux-amd64.tar.gz
+            echo 'export PATH=$PATH:/usr/local/go/bin' >> $HOME/.migration_installer_bashrc
+            echo 'export GOPATH=$HOME/go' >> $HOME/.migration_installer_bashrc
+            echo 'export PATH=$PATH:$GOPATH/bin' >> $HOME/.migration_installer_bashrc
+            echo "Golang Installed."
             break;;
             [Nn]* ) echo "Cannot proceed without go. Exiting....."; exit;;
             * ) echo "Please answer y/Y or n/N.";;
         esac
     done
 else
-    echo "go is already installed. Skipping."
+    GO_INSTALLED_VERSION=$(go version | cut -d " " -f 3)    
+    if [ "$GO_INSTALLED_VERSION" \> "$GO_MIN_VERSION" ] || [ "$GO_INSTALLED_VERSION" = "$GO_MIN_VERSION" ] ; then 
+        echo "$GO_INSTALLED_VERSION is already installed. Skipping."
+    else 
+        echo -e "Installed $GO_INSTALLED_VERSION version is below the minimum required version $GO_MIN_VERSION.\nPlease uninstall current $GO_INSTALLED_VERSION and rerun this scirpt."
+        exit
+    fi
 fi
-source ~/.bashrc
+source $HOME/.migration_installer_bashrc
 
-which pg_dump &>> temp.log
+
+which pg_dump &>> migration_installer.log
 if [ $? -ne 0 ]; then
     echo "pg_dump not found."
     while true; do
-        read -p "Install Postgres11? (Y/N): " yn
+        read -p "Install Postgres14? (Y/N): " yn
         case $yn in
             [Yy]* ) 
-            echo "Installing Postgress11."
-            sudo yum install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-7-x86_64/pgdg-redhat-repo-latest.noarch.rpm >> mylog.log
+            echo "Installing Postgres14"
```

Comment:
Are we sure that we want to install PG14 on the client machine? I'm not sure which versions work with yb_migrate

Reply:
in case its not installed then only we are going to do that. Other we are checking for minimum version which is 10 i think rightnow. PG_DUMP_MIN_VERSION